CMake warnings and errors from glew and glm

For documentation purposes this clarifies why we build and install glew and glm from source rather than using Ubuntu’s default package repositories, specifically when using the nvcr.io/nvidia/cuda:12.5.0-runtime-ubuntu22.04 Docker image.

**glew**

```
2.391 CMake Warning at CSGOptiX/tests/CMakeLists.txt:45 (find_package):
2.391   By not providing "Findglew.cmake" in CMAKE_MODULE_PATH this project has
2.391   asked CMake to find a package configuration file provided by "glew", but
2.391   CMake did not find one.
2.391 
2.391   Could not find a package configuration file provided by "glew" with any of
2.391   the following names:
2.391 
2.391     glewConfig.cmake
2.391     glew-config.cmake
2.391 
2.391   Add the installation prefix of "glew" to CMAKE_PREFIX_PATH or set
2.391   "glew_DIR" to a directory containing one of the above files.  If "glew"
2.391   provides a separate development package or SDK, be sure it has been
2.391   installed.
```

```
3.902 CMake Error at CSGOptiX/tests/CMakeLists.txt:51 (target_link_libraries):
3.902   Target "CSGOptiXRenderInteractiveTest" links to:
3.902 
3.902     GLEW::glew
3.902 
3.902   but the target was not found.  Possible reasons include:
3.902 
3.902     * There is a typo in the target name.
3.902     * A find_package call is missing for an IMPORTED target.
3.902     * An ALIAS target is missing.
```

**glm**

```
1.603 CMake Deprecation Warning at /usr/lib/cmake/glm/glmConfig.cmake:1 (cmake_minimum_required):
1.603   Compatibility with CMake < 3.5 will be removed from a future version of
1.603   CMake.
1.603 
1.603   Update the VERSION argument <min> value or use a ...<max> suffix to tell
1.603   CMake that the project does not need compatibility with older versions.
1.603 Call Stack (most recent call first):
1.603   sysrap/CMakeLists.txt:6 (find_package)
1.603 
1.603 
1.603 CMake Deprecation Warning at /usr/lib/cmake/glm/glmConfig.cmake:2 (cmake_policy):
1.603   Compatibility with CMake < 3.5 will be removed from a future version of
1.603   CMake.
1.603 
1.603   Update the VERSION argument <min> value or use a ...<max> suffix to tell
1.603   CMake that the project does not need compatibility with older versions.
1.603 Call Stack (most recent call first):
1.603   sysrap/CMakeLists.txt:6 (find_package)
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -37,7 +37,7 @@ RUN mkdir -p /opt/glew/src && curl -sL https://github.com/nigels-com/glew/releas
  && cmake --build /opt/glew/build --parallel --target install \
  && rm -fr /opt/glew
 
-RUN mkdir -p /opt/glm/src && curl -sL https://github.com/g-truc/glm/archive/refs/tags/1.0.1.tar.gz | tar -xz --strip-components 1 -C /opt/glm/src
+RUN mkdir -p /opt/glm/src && curl -sL https://github.com/g-truc/glm/archive/refs/tags/1.0.1.tar.gz | tar -xz --strip-components 1 -C /opt/glm/src \
  && cmake -S /opt/glm/src -B /opt/glm/build \
  && cmake --build /opt/glm/build --parallel --target install \
  && rm -fr /opt/glm

--- a/Dockerfile
+++ b/Dockerfile
@@ -17,7 +17,7 @@ RUN apt update \
 RUN apt update \
  && apt install -y libssl-dev \
     nlohmann-json3-dev \
-    libglew-dev libglfw3-dev libglm-dev libglu1-mesa-dev libxmu-dev \
+    libglfw3-dev libglu1-mesa-dev libxmu-dev \
     cmake qtbase5-dev libxerces-c-dev libexpat1-dev \
  && apt clean \
  && rm -rf /var/lib/apt/lists/*
@@ -36,6 +36,11 @@ RUN mkdir -p /opt/glew/src && curl -sL https://github.com/nigels-com/glew/releas
  && cmake -S /opt/glew/src/build/cmake -B /opt/glew/build \
  && cmake --build /opt/glew/build --parallel --target install \
  && rm -fr /opt/glew
+
+RUN mkdir -p /opt/glm/src && curl -sL https://github.com/g-truc/glm/archive/refs/tags/1.0.1.tar.gz | tar -xz --strip-components 1 -C /opt/glm/src
+ && cmake -S /opt/glm/src -B /opt/glm/build \
+ && cmake --build /opt/glm/build --parallel --target install \
+ && rm -fr /opt/glm
 
 RUN mkdir -p /opt/plog/src && curl -sL https://github.com/SergiusTheBest/plog/archive/refs/tags/1.1.10.tar.gz | tar -xz --strip-components 1 -C /opt/plog/src \
  && cmake -S /opt/plog/src -B /opt/plog/build \

